### PR TITLE
fix(crd): add omitempty in spec for crds

### PR DESCRIPTION
**Pull request by rajaSahil**
**Purpose of PR?**:

Fixes #
-  Issues with respect to generation of CR using go-client which was failing because of required fields for `MatchCapabilitiesType` and `MatchProtocols`.
- Add omitempy for the above mentioned fields.

**Does this PR introduce a breaking change?**

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [ ] Bug fix. Fixes #<issue number>
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [X] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: pkg/KubeArmorController/api/security.kubearmor.com/v1/common.go [hunk 1]
```diff
@@ -96,7 +96,7 @@ type ProcessType struct {
 }
 
 type FilePathType struct {
-	Path MatchPathType `json:"path"`
+	Path MatchPathType `json:"path,omitempty"`
```
Comment (daemon1024) [inline pkg/KubeArmorController/api/security.kubearmor.com/v1/common.go]: I don't think this should be `omitempty`. Path is a required field provided matchPaths is used

File: pkg/KubeArmorController/api/security.kubearmor.com/v1/common.go [hunk 2]
```diff
@@ -117,7 +117,7 @@ type FilePathType struct {
 }
 
 type FileDirectoryType struct {
-	Directory MatchDirectoryType `json:"dir"`
+	Directory MatchDirectoryType `json:"dir,omitempty"`
```
Comment (daemon1024) [inline pkg/KubeArmorController/api/security.kubearmor.com/v1/common.go]: Same as path, shouldn't be omitempth

**Review by daemon1024 — CHANGES_REQUESTED** [review]
`matchPaths` is already omitempty so shouldn't cause issues.

File: pkg/KubeArmorController/api/security.kubearmor.com/v1/common.go (hunk 1 above)
Comment (rajaSahil) [reply pkg/KubeArmorController/api/security.kubearmor.com/v1/common.go]: Ah, yes.

File: pkg/KubeArmorController/api/security.kubearmor.com/v1/common.go [hunk 3]
```diff
@@ -297,7 +297,7 @@ type ActionType string
 type Syscall string
 
 type SyscallFromSourceType struct {
-	Path      MatchPathType `json:"path,omitempty"`
+	Path      MatchPathType `json:"path"`
```
Comment (daemon1024) [inline pkg/KubeArmorController/api/security.kubearmor.com/v1/common.go]:
```suggestion
	Path      MatchPathType `json:"path,omitempty"`
```
Path is not mandatory in SysCallFromSource

**Review by daemon1024 — CHANGES_REQUESTED** [review]
Can you rebase your PR from main, we have made fixes to CI

**Review by nyrahul — APPROVED** [review]

**Review by daemon1024 — APPROVED** [review]

**Comment by daemon1024** [comment]
Can you rebase PR again 😓 the tests should be fixed now.

File: pkg/KubeArmorController/config/webhook/manifests.yaml [hunk 4]
```diff
@@ -26,7 +26,3 @@ webhooks:
     resources:
     - pods
   sideEffects: NoneOnDryRun
-  objectSelector:
-    matchExpressions:
-    - key: "kubearmor-app"
-      operator: DoesNotExist
```
Comment (DelusionalOptimist) [inline pkg/KubeArmorController/config/webhook/manifests.yaml:32]: @rajaSahil any reason for removing this filter?

**Review by daemon1024 — APPROVED** [review]

**Comment by anurag-rajawat** [comment]
Can we merge this?

**Pull request merged**